test/test_arraycontext.py

```diff
@@ -738,6 +738,57 @@ def test_norm_ord_none(actx_factory, ndim):
     np.testing.assert_allclose(actx.to_numpy(norm_a), norm_a_ref)
 
 
+def test_outer(actx_factory):
+    actx = actx_factory()
+
+    a_dof, a_ary_of_dofs, _, _, a_bcast_dc_of_dofs = _get_test_containers(actx)
+
+    b_dof = a_dof + 1
+    b_ary_of_dofs = a_ary_of_dofs + 1
+    b_bcast_dc_of_dofs = a_bcast_dc_of_dofs + 1
+
+    from arraycontext import outer
+
+    # Two scalars
+    assert outer(a_dof, b_dof) == a_dof*b_dof
+
+    # Scalar and vector
+    assert np.all(outer(a_dof, b_ary_of_dofs) == a_dof*b_ary_of_dofs)
+    assert np.all(outer(a_ary_of_dofs, b_dof) == a_ary_of_dofs*b_dof)
+
+    # Two vectors
+    assert (
+        np.all(outer(a_ary_of_dofs, b_ary_of_dofs)
+        == np.outer(a_ary_of_dofs, b_ary_of_dofs)))
```

Comment:
Do these `==` do what you intend? (I'm worried about something being array-valued and then just evaluating to True on account of that.)

* I think we'll want a function that checks structural equality in the presence of array containers.
* We should make sure that array container behavior is safe when it comes to equality comparison.